gravity-forms/gw-gravity-forms-rename-uploaded-files.php

```diff
@@ -241,32 +241,28 @@ function is_path( $filename ) {
 		return strpos( $filename, '/' ) !== false;
 	}
 
-	function get_template_value( $template, $file, $entry ) {
-
+	function get_renamed_filepath( $template, $file, $entry ) {
 		$info = pathinfo( $file );
 
-		if ( strpos( $template, '/' ) === 0 ) {
-			$dir      = wp_upload_dir();
-			$template = $dir['basedir'] . $template;
-		} else {
-			$template = $info['dirname'] . '/' . $template;
-		}
-
 		// replace our custom "{filename}" psuedo-merge-tag
-		$value = str_replace( '{filename}', $info['filename'], $template );
+		$filename = str_replace( '{filename}', $info['filename'], $template );
 
 		// replace merge tags
-		$form  = GFAPI::get_form( $entry['form_id'] );
-		$value = GFCommon::replace_variables( $value, $form, $entry, false, true, false, 'text' );
+		$form     = GFAPI::get_form( $entry['form_id'] );
+		$filename = GFCommon::replace_variables( $template, $form, $entry, false, true, false, 'text' );
```

Comment:
The biggest change here is that the merge tags are now replaced before concatenating the user-inputted filename with the rest of the file path.

This allows us to clean/sanitize the filename completely before it's added to the file path whereas adding it to the file path first, before cleaning, makes it impossible to know if all user input has been cleaned.